src/loaders/IVolumeLoader.ts

```diff
@@ -65,3 +64,60 @@ export interface IVolumeLoader {
   // TODO explicitly passing a `LoadSpec` is now rarely useful. Remove?
   loadVolumeData(volume: Volume, loadSpec?: LoadSpec, onChannelLoaded?: PerChannelCallback): void;
 }
+
+/** Abstract class which allows loaders to accept and return types that are easier to transfer to/from a worker. */
+export abstract class ThreadableVolumeLoader implements IVolumeLoader {
+  /** Unchanged from `IVolumeLoader`. See that interface for details. */
+  abstract loadDims(loadSpec: LoadSpec): Promise<VolumeDims[]>;
+
+  /**
+   * Creates an `ImageInfo` object from a `LoadSpec`, which may be passed to the `Volume` constructor to create an
+   * empty volume that can accept data loaded with the given `LoadSpec`.
+   *
+   * Also returns a new `LoadSpec` that may have been modified from the input `LoadSpec` to reflect the constraints or
+   * abilities of the loader. This new `LoadSpec` should be used when constructing the `Volume`, _not_ the original.
+   */
+  abstract createImageInfo(loadSpec: LoadSpec): Promise<[ImageInfo, LoadSpec]>;
+
+  /**
+   * Begins loading per-channel data for the volume specified by `imageInfo` and `loadSpec`.
+   *
+   * Returns a promise that resolves to reflect any modifications to `imageInfo` and/or `loadSpec` that need to be made
+   * based on this load. Actual loaded channel data is passed to `onData` as it is loaded. Depending on the format,
+   * the returned array may be in simple 3d dimension order or reflect a 2d atlas. If the latter, the dimensions of the
+   * atlas are passed as the third argument to `onData`.
+   */
+  abstract loadRawChannelData(
+    imageInfo: ImageInfo,
+    loadSpec: LoadSpec,
+    onData: RawChannelDataCallback
+  ): Promise<[ImageInfo | undefined, LoadSpec | undefined]>;
```

Comment:
Is this function only called internally?  Could it be private?

Reply:
> would it make any sense for the promise type to include the arguments to RawChannelDataCallback? Or will the ImageInfo,Loadspec resolve at a different time than the onData callback is called?

I think I now understand that this function is also potentially loading multiple channels and needs to fire the callback once per channel received.

Reply:
At least in the case of `OMEZarrLoader`, this function isn't `async` and returns a promise that resolves immediately.

Reply:
> I think I now understand that this function is also potentially loading multiple channels and needs to fire the callback once per channel received.

Right - this is basically supposed to be a stripped-down equivalent to `loadVolumeData`, so the callback argument works the same as it would there.

Reply:
> Is this function only called internally? Could it be private? (I forget if ts has a notion of "protected")

The intent of this function is to be a version of `loadVolumeData` that doesn't need access to the `Volume` it's working on, so it's called externally when a loader is running on a worker. It just so happened that all volume loaders modified `Volume` in a similar enough way to also allow a common implementation of `loadVolumeData` based around it.

Reply:
Makes me wonder if this could just be the new IVolumeLoader interface and not need the layer of ThreadableVolumeLoader

Reply:
I'm 50/50 on doing that. On the one hand, `IVolumeLoader` still represents a smaller set of requirements and there's a genuine distinction between when a property or method should be typed with one vs. the other. On the other hand, I have a hard time imagining an `IVolumeLoader` that couldn't be implemented via `ThreadableVolumeLoader` - i.e., one that would want to make changes other than those permitted by `ThreadableVolumeLoader` (modify `ImageInfo` more than once, say, or call some method on a `Volume`). 

Maybe the cleanest way from a types perspective would be to convert the abstract methods of `ThreadableVolumeLoader` to the new `IVolumeLoader` interface and move the `Volume`-changing code currently implemented within `ThreadableVolumeLoader` somewhere else? But it's not super obvious to me where it might go instead.

Reply:
Lets leave this for later discussion... I thought I was noticing that everything is now a ThreadableVolumeLoader and there are no pure IVolumeLoaders.